### DISCUSSION TITLE
The openthread protocol stack cannot be networked after upgrading from the June version of this year to the current new version

### DISCUSSION THREAD

**Discussion by tuopu** [openthread/openthread#9426] [Q&A]
Hello experts

The version I am using is the upgraded version from June 2023, and the networking is normal. I am currently upgrading to the latest version, but the problem I encountered is that I cannot network anymore. The file compilation is normal, and the network status of any node is OT_ DIVICE_ ROLE_ The DECHED state, logically speaking, should be OT after each node enables network connection if communication is not possible in a timely manner_ DIVICE_ ROLE_ LEADER status, could you please provide expert tips on which settings need to be added or changed in the new version? Thank you very much.

**Comment by jwhui** [comment]
Can you provide more information?
1. OpenThread git commits are you using before and after the upgrade
2. Logs and/or packet sniffer traces

**Comment by tuopu** [comment]
Query the June version using the git log as follows：


commit 80eb7f512fd05f610d94d1c1c73e647f2a660e1a (HEAD -&gt; main, origin/main, origin/HEAD)
Author: Abtin Keshavarzian ***@***.***&gt;
Date:&nbsp; &nbsp;Thu Jun 1 21:39:45 2023 -0700


&nbsp; &nbsp; [toranj] update `build.sh` script to use CMake for all builds (#9115)


commit 9bff9f0807a06b4c2634d5bf14bd92e288ab9164
Author: Abtin Keshavarzian ***@***.***&gt;
Date:&nbsp; &nbsp;Thu Jun 1 17:49:50 2023 -0700


&nbsp; &nbsp; [routing-manager] fix deprecating on-link prefix from inactive router (#9121)


&nbsp; &nbsp; This commit fixes `RemoveOrDeprecateEntriesFromInactiveRouters()`
&nbsp; &nbsp; to ensure to skip over an already deprecating on-link prefix (and
&nbsp; &nbsp; not remove it).


commit 3eb48ca00c425d62a866e894f40579cf5160235a
Author: Abtin Keshavarzian ***@***.***&gt;
Date:&nbsp; &nbsp;Thu Jun 1 13:05:06 2023 -0700


&nbsp; &nbsp; [backbone-router] use `Clamp` to check MLR timeout value (#9119)
:...skipping...
commit 80eb7f512fd05f610d94d1c1c73e647f2a660e1a (HEAD -&gt; main, origin/main, origin/HEAD)
Author: Abtin Keshavarzian ***@***.***&gt;
Date:&nbsp; &nbsp;Thu Jun 1 21:39:45 2023 -0700


&nbsp; &nbsp; [toranj] update `build.sh` script to use CMake for all builds (#9115)


commit 9bff9f0807a06b4c2634d5bf14bd92e288ab9164
Author: Abtin Keshavarzian ***@***.***&gt;
Date:&nbsp; &nbsp;Thu Jun 1 17:49:50 2023 -0700


&nbsp; &nbsp; [routing-manager] fix deprecating on-link prefix from inactive router (#9121)


&nbsp; &nbsp; This commit fixes `RemoveOrDeprecateEntriesFromInactiveRouters()`
&nbsp; &nbsp; to ensure to skip over an already deprecating on-link prefix (and
&nbsp; &nbsp; not remove it).


commit 3eb48ca00c425d62a866e894f40579cf5160235a
Author: Abtin Keshavarzian ***@***.***&gt;
Date:&nbsp; &nbsp;Thu Jun 1 13:05:06 2023 -0700


&nbsp; &nbsp; [backbone-router] use `Clamp` to check MLR timeout value (#9119)


commit 87e3250a51b8c21ceb33ce544c5cebf9b0ba070d
Author: Abtin Keshavarzian ***@***.***&gt;
Date:&nbsp; &nbsp;Thu Jun 1 13:04:09 2023 -0700


&nbsp; &nbsp; [core] use `Min()`, `Max()` to clip values (#9117)


commit 8f5d4376c6b5c681ca3c426feb477825bc7dbfad
Author: Abtin Keshavarzian ***@***.***&gt;
Date:&nbsp; &nbsp;Wed May 31 13:26:39 2023 -0700


&nbsp; &nbsp; [build] delete the removed child supervision config in `cmake-build` (#9111)


commit 9c7e679ed16e8d544e29dcd085bcb0cf5415d4d1
Author: Song GUO ***@***.***&gt;
Date:&nbsp; &nbsp;Wed May 31 14:16:43 2023 +0800


&nbsp; &nbsp; [posix] add unified platform API for setting NAT64 CIDR during runtime (#8947)


&nbsp; &nbsp; We should notice that the default CIDR (`192.168.255.0/24`) is not
&nbsp; &nbsp; safe to use, however, in most cases, the CIDR used can only be
&nbsp; &nbsp; determined during runtime since the single binary might be distributed
&nbsp; &nbsp; to BRs in different network conditions. And we cannot conclude an
&nbsp; &nbsp; always safe CIDR in all conditions.


&nbsp; &nbsp; This commit will emit an event when NAT64 CIDR is changed. So the
&nbsp; &nbsp; platform driver can update the route for NAT64.

**Comment by jwhui** [comment]
My guess is that you are not configuring the Operational Dataset properly. I suggest reading [src/cli/README_DATASET.md](https://github.com/openthread/openthread/blob/main/src/cli/README_DATASET.md) for more information on properly configuring the Operational Dataset.
